### PR TITLE
Add ProjectsManager to keep track of version info

### PR DESCRIPTION
**Pull request by cdonati**
This allows the AppManager to retrieve version details without the AppController's input.

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/3715/

File: AdminServer/appscale/admin/instance_manager/projects_manager.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        service IDs.
+    """
+    if self._stopped:
+      return False
+
+    main_io_loop = IOLoop.instance()
+    main_io_loop.add_callback(self.update_services, new_services_list)
+
+
+class ProjectsManager(dict):
```
Comment (whoarethebritons) [inline AdminServer/appscale/admin/instance_manager/projects_manager.py]: can this be changed to `GlobalProjectsManager` it seems to easy too confuse with the ProjectManager

File: AppManager/app_manager_server.py [hunk 2]
```diff
@@ -381,7 +370,7 @@ def stop_app_instance(app_name, port):
     return False
 
   logging.info("Stopping application %s" % app_name)
-  watch = "app___" + app_name + "-" + str(port)
+  watch = '{}{}-{}'.format(MONIT_INSTANCE_PREFIX, app_name, port)
```
Comment (whoarethebritons) [inline AppManager/app_manager_server.py]: how come this one is a format but the previous and next ones are ''.join

File: AdminServer/appscale/admin/instance_manager/projects_manager.py (hunk 1 above)
Comment (cdonati) [inline AdminServer/appscale/admin/instance_manager/projects_manager.py]: Yes, I can change it.

**Pull request merged**